core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/saved-workflow-section.component.ts

```diff
@@ -270,24 +271,32 @@ export class SavedWorkflowSectionComponent implements OnInit, OnChanges {
   /**
    * Search workflows based on date string
    * String Formats:
-   *  - ctime:YYYY-MM-DD (workflows on this date)
-   *  - ctime:<YYYY-MM-DD (workflows on or before this date)
-   *  - ctime:>YYYY-MM-DD (workflows on or after this date)
+   *  - mtime:YYYY-MM-DD (workflows on this date)
+   *  - mtime:<YYYY-MM-DD (workflows on or before this date)
+   *  - mtime:>YYYY-MM-DD (workflows on or after this date)
    */
-  private searchCreationTime(
-    date: Date,
-    filteredDashboardWorkflowEntries: ReadonlyArray<DashboardWorkflowEntry>
+
+  private searchDate(
+    date: Date[],
+    filteredDashboardWorkflowEntries: ReadonlyArray<DashboardWorkflowEntry>,
+    type: String
   ): ReadonlyArray<DashboardWorkflowEntry> {
-    date.setHours(0), date.setMinutes(0), date.setSeconds(0), date.setMilliseconds(0);
+    // eslint-disable-next-line no-unused-expressions
+    date[0].setHours(0),
+      date[0].setMinutes(0),
+      date[0].setSeconds(0),
+      date[0].setMilliseconds(0),
+      date[1].setHours(0),
+      date[1].setMinutes(0),
+      date[1].setSeconds(0),
+      date[1].setMilliseconds(0);
     //sets date time at beginning of day
     //date obj from nz-calendar adds extraneous time
     return filteredDashboardWorkflowEntries.filter(workflow_entry => {
       //filters for workflows that were created on the specified date
-      if (workflow_entry.workflow.creationTime) {
-        return (
-          workflow_entry.workflow.creationTime >= date.getTime() &&
-          workflow_entry.workflow.creationTime < date.getTime() + 86400000
-        );
+      let time = type == "C" ? workflow_entry.workflow.creationTime : workflow_entry.workflow.lastModifiedTime;
```

Comment:
Try to use "===" and "!==" instead of "==" and "!=". Same applies to other places here.